libfreerdp/core/connection.c

```diff
@@ -437,21 +435,19 @@ BOOL rdp_client_reconnect(rdpRdp* rdp)
 	BOOL status;
 	rdpContext* context;
 	rdpChannels* channels;
-	BOOL connected;
 
 	if (!rdp || !rdp->context || !rdp->context->channels)
 		return FALSE;
 
 	context = rdp->context;
 	channels = context->channels;
-	connected = channels->connected;
 
 	if (!rdp_client_disconnect_and_clear(rdp))
 		return FALSE;
 
 	status = rdp_client_connect(rdp);
 
-	if (status && connected)
+	if (status && (context->instance->ConnectionCallbackState == 2))
 		status = (freerdp_channels_post_connect(channels, context->instance) == CHANNEL_RC_OK);
```

Comment:
Not your code, but wouldn't it be nice to have some enum instead of raw values for ConnectionCallbackState ?